# 57) Tek Örneklem T Testi
Tek Örneklem T Testi, Z Testi ile **birebir aynı amacı** taşır. Bir örneklem ortalamasının, iddia edilen bir değere eşit olup olmadığını test eder. Tek fark, **hangi koşullarda kullanıldığı.**

## Ne Zaman Z Yerine T Kullanılır?
- Popülasyonun standart sapması ($\sigma$) **bilinmiyorsa**, VE
- Örneklem büyüklüğü **n < 30** ise

Bu durumda, örneklemden hesapladığımız $s$ (std) ile $\sigma$'yı tahmin etmemiz gerekir bu ekstra belirsizlik, t-dağılımının (normalden daha kalın kuyruklu) kullanılmasını gerektirir.

## Test İstatistiği Formülü
$$t = \frac{\bar{x} - \mu_0}{s / \sqrt{n}}$$

Z formülüyle (Konu 57) **birebir aynı görünüyor** tek fark, paydadaki $\sigma$ yerine örneklemden hesapladığımız $s$ kullanılması ama bu küçük fark, sonucu **hangi dağılıma göre** yorumlayacağımızı tamamen değiştiriyor.

## Serbestlik Derecesi
T testinde, serbestlik derecesi $df = n-1$'dir, t-dağılımının şeklini belirleyen parametre bu.

## P-Değeri Hesabı Farkı
Z testinde `stats.norm.cdf()` kullanıyorduk (normal dağılım). T testinde ise `stats.t.cdf()` kullanırız ve **df parametresini** de vermemiz gerekir (t-dağılımının şekli df'e bağlı olduğu için).

## Örnek Senaryo
Bir kahve dükkanı, bir fincan kahvenin ortalama hazırlanma süresinin **3 dakika** olduğunu iddia ediyor. Popülasyon std'si bilinmiyor. **12 fincanlık** küçük bir örnekleme sahibiz (n=12, n<30 olduğu için T testi gerekiyor), ortalama süre **3.4 dakika**, örneklem std'si **0.6 dakika** çıktı. Süre gerçekten iddia edilenden farklı mı (çift kuyruklu)?

In [2]:
from scipy import stats
from math import sqrt
# H0: bir fincan kahvenin ortalama hazırlanma süresi 3 dakikadır.
# H1: bir fincan kahvenin ortalama hazırlanma süresi 3 dakikadan farklıdır. 
pop_ortalamasi = 3
orneklem_ortalamasi = 3.4
orneklem_sapmasi = 0.6
n = 12
t_istatistigi = (orneklem_ortalamasi - pop_ortalamasi) / (orneklem_sapmasi / sqrt(n))

df = n - 1  # serbestlik derecesi
p_degeri = 2 * stats.t.cdf(-abs(t_istatistigi), df=df)
alpha = 0.05
print(f'T İstatistiği: {t_istatistigi}')
if p_degeri < alpha:
    print(f'P Değeri: {p_degeri}, H0 reddedilir. Bir fincan kahvenin ortalama hazırlama süresi 3 dakikadan farklıdır.')
else:
    print(f'P Değeri: {p_degeri}, H0 reddedecek kadar kanıt yok elimizde. Bir fincan kahvenin ortalama hazırlama süresi 3 dakikadır.')

T İstatistiği: 2.3094010767585025
P Değeri: 0.04134226737172354, H0 reddedilir. Bir fincan kahvenin ortalama hazırlama süresi 3 dakikadan farklıdır.


### Sonuç
12 fincanlık örneklemden elde edilen ortalama hazırlama süresi (3.4 dakika), iddia edilen 3 dakikadan istatistiksel olarak anlamlı şekilde farklı bulunmuştur (t(11)=2.31, p=0.041). Bu doğrultuda kahve dükkanının, hazırlama sürecini neden iddia edilenden farklı sürdüğünü araştırması önerilir.